src/app/pages/CreateConnectorPage/components/ConnectorFilter.tsx

```diff
@@ -0,0 +1,118 @@
+import { validateConnectorSearchField } from '@utils/shared';
+import React, { FC, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  Button,
+  ButtonVariant,
+  InputGroup,
+  TextInput,
+  Tooltip,
+  ValidatedOptions,
+  TextInputGroup,
+  TextInputGroupUtilities,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
+
+export type ConnectorFilter = {
+  onChangeSearchField: (value: string) => void;
+  loading: boolean;
+  searchFieldPlaceholder: string;
+  total: number;
+  searchFieldValue: string;
+};
+export const ConnectorFilter: FC<ConnectorFilter> = ({
+  onChangeSearchField,
+  loading,
+  searchFieldPlaceholder,
+  searchFieldValue,
+}) => {
+  const { t } = useTranslation();
+  const [valid, setValid] = useState<boolean>(true);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [currentSearch, setCurrentSearch] = useState<string | undefined>(
+    searchFieldValue
+  );
+  const [showClearButton, SetShowClearButton] = useState<boolean>(false);
+
+  const onChange = (input?: string) => {
+    setCurrentSearch(input);
+    setValid(validateConnectorSearchField(input));
+  };
+
+  const onFilter = () => {
+    if (currentSearch && currentSearch.trim() != '') {
+      if (validateConnectorSearchField(currentSearch)) {
+        onChangeSearchField(currentSearch);
+        setCurrentSearch('');
+      } else {
+        setValid(false);
+      }
+    }
+    SetShowClearButton(true);
+  };
+  const renderInput = () => {
+    const FilterTooltip: FC = () => {
```

Comment:
What advantage does this bring over this being inline on line 113?  Apologies if I've missed that this component is used in a couple spots, I'm just curious on this pattern :smile:

Reply:
ok cool, I had thought probably I didn't spot the other place where this was used.  Thanks!